nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -877,6 +878,120 @@ impl<'a> Planner<'a> {
         Ok(())
     }
 
+    /// Update at most one existing zone to use a new image source.
+    fn do_plan_zone_updates(&mut self) -> Result<(), Error> {
+        // We are only interested in non-decommissioned sleds.
+        let sleds = self
+            .input
+            .all_sleds(SledFilter::Commissioned)
+            .map(|(id, _details)| id)
+            .collect::<Vec<_>>();
+
+        // Wait for all current zones to appear in the inventory.
+        // It would be nice if we could check their image source,
+        // but the inventory doesn't report that correctly: see
+        // <https://github.com/oxidecomputer/omicron/issues/8084>.
```

Comment:
Not asking for any changes but just checking my understanding: fixing #8084 is more than just a nice improvement - it's actually required for correctness, right? Because the planner here assumes if a zone is present in inventory then its image source matches what the blueprint says it should be?

Reply:
Yes, you are quite correct. Clarified comment in e31fc6098.